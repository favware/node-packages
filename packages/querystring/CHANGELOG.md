# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [4.1.3](https://github.com/favware/node-packages/compare/@favware/querystring@4.1.2...@favware/querystring@4.1.3) (2019-11-09)

**Note:** Version bump only for package @favware/querystring





## [4.1.2](https://github.com/favware/node-packages/compare/@favware/querystring@4.1.1...@favware/querystring@4.1.2) (2019-10-13)


### Bug Fixes

* completely re-do tsconfig standardized ([9eb030f](https://github.com/favware/node-packages/commit/9eb030fdf1deb75d5ae8b273d0e9c359bcb985a1))





## [4.1.1](https://github.com/favware/node-packages/compare/@favware/querystring@4.1.0...@favware/querystring@4.1.1) (2019-10-13)

**Note:** Version bump only for package @favware/querystring





# [4.1.0](https://github.com/favware/node-packages/compare/@favware/querystring@4.0.0...@favware/querystring@4.1.0) (2019-10-13)


### Features

* standardize TS configuration ([47696f4](https://github.com/favware/node-packages/commit/47696f4e1dd2632b305ff9789cdd6c473fa709ca))





# [4.0.0](https://github.com/favware/node-packages/compare/@favware/querystring@3.0.3...@favware/querystring@4.0.0) (2019-09-11)


### Features

* **querystring:** completely ignore "undefined" and "null" ([c8620dc](https://github.com/favware/node-packages/commit/c8620dc))


### BREAKING CHANGES

* **querystring:** Previously there would be "prop=undefined", now the entire prop is gone.





## [3.0.3](https://github.com/favware/node-packages/compare/@favware/querystring@3.0.2...@favware/querystring@3.0.3) (2019-09-11)

**Note:** Version bump only for package @favware/querystring





## [3.0.2](https://github.com/favware/node-packages/compare/@favware/querystring@3.0.1...@favware/querystring@3.0.2) (2019-08-21)

**Note:** Version bump only for package @favware/querystring





## 3.0.1 (2019-08-18)


### Reverts

* "ci: another attempt to fix CI config" ([37c385f](https://github.com/favware/node-packages/commit/37c385f))
