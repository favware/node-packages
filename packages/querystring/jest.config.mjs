/** @type {import('@jest/types').Config.InitialOptions} */
const config = {
  displayName: 'ts-jest',
  preset: 'ts-jest',
  globals: {
    'ts-jest': {
      tsconfig: '<rootDir>/__tests__/tsconfig.json'
    }
  }
};

export default config;
