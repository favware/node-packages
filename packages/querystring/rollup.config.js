import configure from '../../rollup.config';
import { terser } from 'rollup-plugin-terser';

export default configure({
   input: 'src/index.ts',
   plugins: [
    terser({ ecma: 5 }),
   ]
});