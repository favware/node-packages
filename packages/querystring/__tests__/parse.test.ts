import { parse } from '../src/querystring';
import { oneLine } from 'common-tags';

describe('Standard Behaviour', () => {
  test('should work with empty input', () => {
    const expected = {};
    const actual = parse('');
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);
  });

  test('should work without options', () => {
    const expected = { prop: 'value', prop2: 'value2' };
    const actual = parse('?prop=value&prop2=value2');
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);
  });

  test('should work when key exists multiple times', () => {
    const expected = { prop: [ 'value', 'value' ] };
    const actual = parse('?prop=value&prop=value');
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);
  });

  test('should work when key has array-like value', () => {
    const expected = { prop: '[\'value\', \'value\']' };
    const actual = parse('?prop=[\'value\', \'value\']');
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);
  });

  test('should work with no parameters', () => {
    const expected = {};
    const actual = parse();
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);
  });

  test('should work with valid options', () => {
    const expected = { prop: 'value', prop2: 'value2' };
    const actual = parse('?prop=value&prop2=value2', { separator: '&', equals: '=' });
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);
  });

  test('should work when passing a full URL', () => {
    const expected = { prop: 'value', prop2: 'value2' };
    const actual = parse('https://google.com/?prop=value&prop2=value2', { separator: '&', equals: '=' });
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);
  });

  test('should work when exceeding max keys', () => {
    /* eslint-disable max-len */
    const keys = oneLine`
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    prop=value&prop0=value&prop1=value&prop2=value&prop3=value&prop4=value&prop5=value&prop6=value&prop7=value&prop8=value&prop9=value&prop10=value&prop11=value&prop12=value&prop13=value&prop14=value&prop15=value&prop16=value&prop17=value&prop18=value&prop19=value&prop20=value&
    `.replace(/ /gm, '');
    const expected = { prop: [ 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value' ], prop2: [ 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value', 'value' ] };
    /* eslint-enable max-len */
    const actual = parse(`?${keys}`, { separator: '&', equals: '=' });
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);

    expect(true).toBe(true);
  });
});

describe('Error checking', () => {
  /* eslint-disable @typescript-eslint/ban-ts-ignore */
  test('should work with wrong options', () => {
    const expected = { prop: 'value', prop2: 'value2' };
    // @ts-ignore
    const actual = parse('?prop%value&prop2%value2', { seperator: '&', equals: '%' });
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);
  });

  test('should error with wrong input', () => {
    const expected = { err: '@Favware/Querystring: Your input was not an string. Please supply a string when using Parse' };
    // @ts-ignore
    const actual = parse({ prop: 'value', prop2: 'value2' });
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);
  });

  test('should work when passing invalid separator and equals', () => {
    const expected = { prop: 'value', prop2: 'value2' };
    // @ts-ignore
    const actual = parse('?prop=value&prop2=value2', { separator: null, equals: null });
    expect(actual).toMatchSnapshot();
    expect(actual).toMatchObject(expected);
  });
  /* eslint-enable @typescript-eslint/ban-ts-ignore */
});