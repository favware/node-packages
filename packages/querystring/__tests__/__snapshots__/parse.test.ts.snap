// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error checking should work when passing invalid separator and equals 1`] = `
Object {
  "prop": "value",
  "prop2": "value2",
}
`;

exports[`Standard Behaviour should work when exceeding max keys 1`] = `
Object {
  "prop": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop0": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop1": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop10": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop11": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop12": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop13": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop14": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop15": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop16": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop17": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop18": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop19": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop2": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop20": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop3": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop4": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop5": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop6": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop7": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop8": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
  "prop9": Array [
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
    "value",
  ],
}
`;

exports[`Standard Behaviour should work when key exists multiple times 1`] = `
Object {
  "prop": Array [
    "value",
    "value",
  ],
}
`;

exports[`Standard Behaviour should work when key has array-like value 1`] = `
Object {
  "prop": "['value', 'value']",
}
`;

exports[`Standard Behaviour should work when passing a full URL 1`] = `
Object {
  "prop": "value",
  "prop2": "value2",
}
`;

exports[`Standard Behaviour should work with empty input 1`] = `Object {}`;

exports[`Standard Behaviour should work with no parameters 1`] = `Object {}`;

exports[`Standard Behaviour should work with valid options 1`] = `
Object {
  "prop": "value",
  "prop2": "value2",
}
`;

exports[`Standard Behaviour should work without options 1`] = `
Object {
  "prop": "value",
  "prop2": "value2",
}
`;
